asapdiscovery-docking/asapdiscovery/docking/scripts/run_docking_oe.py

```diff
@@ -600,26 +695,35 @@ def main():
                     break
                 except TimeoutError:
                     # This compound:xtal combination timed out
-                    print("Docking timed out for", args_list[8], flush=True)
+                    logger.error("Docking timed out for", args_list[8])
                     failed_runs += [args_list[8]]
                 except pebble.ProcessExpired as e:
-                    print("Docking failed for", args_list[8], flush=True)
-                    print(f"\t{e}. Exit code {e.exitcode}", flush=True)
+                    logger.error("Docking failed for", args_list[8])
+                    logger.error(f"\t{e}. Exit code {e.exitcode}")
                     failed_runs += [args_list[8]]
                 except Exception as e:
-                    print(
-                        "Docking failed for",
-                        args_list[8],
-                        "with Exception",
-                        e,
-                        flush=True,
+                    logger.error(
+                        "Docking failed for", args_list[8], "with Exception", e
                     )
-                    print(e.traceback, flush=True)
+                    logger.error(e.traceback)
                     failed_runs += [args_list[8]]
-            print(f"Docking failed for {len(failed_runs)} runs", flush=True)
+
+                # things are going poorly, lets stop
+                if len(failed_runs) > args.max_failures:
+                    raise ValueError(
+                        f"Too many failures ({len(failed_runs)}/{args.max_failures}) while running docking, aborting"
+                    )
```

Comment:
I think if we call `res.cancel()` instead of just raising the exception, we can then continue to iterate through and collect results for anything that's already finished. not positive though

Reply:
Yep great idea.